modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/prepare/IgniteTypeCoercion.java

```diff
@@ -112,11 +89,6 @@ private boolean doBinaryComparisonCoercion(SqlCallBinding binding) {
 
         validateBinaryComparisonCoercion(binding, leftType, rightType, (IgniteSqlValidator) validator);
 
-        // If types are equal, no need in coercion.
-        if (leftType.equals(rightType)) {
-            return false;
-        }
-
         // Otherwise find the least restrictive type among the operand types
```

Comment:
Why previous part for equals types has been deleted - it looked like as fast path for such a case.
Even if it was a garbage need to amend the comment, due to it's strange to see 'Otherwise bla-bla-bla'.